manager/eris-mint/eris-mint.go

```diff
@@ -185,15 +185,14 @@ func (app *ErisMint) Commit() (res tmsp.Result) {
 	// sync the AppendTx cache
 	app.cache.Sync()
 
-	// if there were any txs in the block,
-	// reset the check cache to the new height
-	if app.nTxs > 0 {
-		log.WithFields(log.Fields{
-			"txs": app.nTxs,
-		}).Info("Reset checkCache")
-		app.checkCache = sm.NewBlockCache(app.state)
-	}
-	app.nTxs = 0
+	// Refresh checkCache even if there were no transactions. A CheckTx may have
```

Comment:
Mulling on it a bit more; checkCache is okay to be ahead of the real application state; in addition to the above remark that we need to strongly enforce the concurrency from within erismint, CheckTx should be more strongly aware of possible of its inability to acknowledge the ordering and as such can be made smarter on explicit validations of ordering; eg sequence number on CheckTx should be a strict inequality, rather than exactly equal to `sequence_number + 1`